### PR TITLE
Migrate setuptools

### LINKED ISSUE #90: Migrate away from setuptools to Poetry
We currently have some files left over from our conversion. The following files can (mostly) be replaced by attributes in pyproject.toml:

- [ ] MANIFEST.in
- [ ] setup.cfg
- [ ] setup.py

We may need to keep setup.cfg around for configuration of tools that do not yet support pyproject.toml. pytest doesn't [yet](https://github.com/pytest-dev/pytest/issues/1556#issuecomment-572267167) and I don't think we can remove the sqla_testing section either.

Looks like the SQLAlchemy entry points can be set using "plugins": https://github.com/python-poetry/poetry/issues/658#issuecomment-467952140

### PR DESCRIPTION
**Pull request by naveen-ram**
Fixes #90

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,34 @@
 [tool.poetry]
 name = "sqlalchemy-ibmi"
-version = "0.1.0"
-description = "'SQLAlchemy support for IBM i db2'"
-authors = ["Naveen-Ram <naveen.ram@ibm.com>"]
+version = "0.9.0"
```
Comment (kadler) [inline pyproject.toml:3]:
```suggestion
version = "0.9.0-dev"
```

.bumpversion.cfg also needs to be updated to include this file now.

File: pyproject.toml [hunk 2]
```diff
@@ -1,8 +1,34 @@
...
+description = "SQLAlchemy support for Db2 on IBM i"
+readme = "README.md"
+author = [
```
Comment (kadler) [inline pyproject.toml:6]:
```suggestion
authors = [
```

https://python-poetry.org/docs/pyproject/#authors

File: pyproject.toml [hunk 3]
```diff
@@ -1,8 +1,34 @@
...
+    "Naveen Ram <naveen.ram@ibm.com>"
+    "Kevin Adler <kadler@us.ibm.com>"
+]
+license = "Apache-2.0"
+keywords = ["sqlalchemy", "database", "ibm", "ibmi", "db2"]
+classifiers =
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: Apache License 2.0",
```
Comment (kadler) [inline pyproject.toml:15]:
```suggestion
```

https://python-poetry.org/docs/pyproject/#classifiers

FYI Poetry automatically adds this from the license field, so it's not needed.

File: pyproject.toml [hunk 4]
```diff
@@ -1,8 +1,34 @@
...
+    {include = "sqlalchemy_ibmi"}
+]
+include = [
+    { path = "test/*", format = "sdist" },
+    { path = "contributing/*", format = "sdist" },
+    { path = "docs/*", format = "sdist" },
+    { path = "setup.cfg", format = "sdist" },
+    { path = "CHANGES", format = "sdist" },
+    { path = "LICENSE", format = "sdist" },
+]
```
Comment (kadler) [inline pyproject.toml:28]: Just an FYI, this requires an unreleased version of Poetry. See https://github.com/IBM/python-itoolkit/commit/64a603df251ad51eb2dd94ba8d5e61debd28d576

Comment (kadler) [reply pyproject.toml:28]: This also requires changing the build-system requires and build-backend to work correctly as poetry is being re-organized to work with PEP 517.

File: setup.cfg [hunk 5]
```diff
@@ -7,33 +7,4 @@ python_files=test/*test_*.py
...
-include_package_data = True
-install_requires =
-    sqlalchemy>=1.3
-    pyodbc>=4.0
-packages = sqlalchemy_ibmi
-
-[options.entry_points]
-sqlalchemy.dialects =
-    ibmi = sqlalchemy_ibmi.base:IBMiDb2Dialect
+profile_file=test/profiles.txt
```
Comment (kadler) [inline setup.cfg:10]: Missing newline

File: pyproject.toml [hunk 6]
```diff
@@ -1,8 +1,34 @@
...
+version = "0.9.0"
+description = "SQLAlchemy support for Db2 on IBM i"
+readme = "README.md"
+author = [
+    "Naveen Ram <naveen.ram@ibm.com>"
+    "Kevin Adler <kadler@us.ibm.com>"
+]
+license = "Apache-2.0"
+keywords = ["sqlalchemy", "database", "ibm", "ibmi", "db2"]
+classifiers =
```
Comment (kadler) [inline pyproject.toml:12]: Looks like this is missing brackets around the list

File: pyproject.toml [hunk 7]
```diff
@@ -1,8 +1,34 @@
 [tool.poetry]
 name = "sqlalchemy-ibmi"
-version = "0.1.0"
-description = "'SQLAlchemy support for IBM i db2'"
-authors = ["Naveen-Ram <naveen.ram@ibm.com>"]
+version = "0.9.0"
+description = "SQLAlchemy support for Db2 on IBM i"
+readme = "README.md"
+author = [
+    "Naveen Ram <naveen.ram@ibm.com>"
```
Comment (kadler) [inline pyproject.toml:7]:
```suggestion
    "Naveen Ram <naveen.ram@ibm.com>",
```

**Comment by kadler** [comment]
Please merge or rebase the README changes from master as well.

File: pyproject.toml (hunk 4 above)
Comment (naveen-ram) [reply pyproject.toml:28]: Should this be left out for now then?

Comment (kadler) [reply pyproject.toml:28]: Nah, we can leave it in. We just might not be able to use automation at the moment to release packages.

File: pyproject.toml [hunk 8]
```diff
@@ -1,8 +1,34 @@
...
 name = "sqlalchemy-ibmi"
-version = "0.1.0"
-description = "'SQLAlchemy support for IBM i db2'"
-authors = ["Naveen-Ram <naveen.ram@ibm.com>"]
+version = "0.9.0-dev"
+description = "SQLAlchemy support for Db2 on IBM i"
+readme = "README.md"
+authors = [
+    "Naveen Ram <naveen.ram@ibm.com>,
+    Kevin Adler <kadler@us.ibm.com>"
```
Comment (kadler) [inline pyproject.toml:8]:
```suggestion
    "Naveen Ram <naveen.ram@ibm.com>",
    "Kevin Adler <kadler@us.ibm.com>"
```

File: pyproject.toml (hunk 4 above)
Comment (kadler) [reply pyproject.toml:28]: We do need to adjust the build-system setting, though.

**Comment by naveen-ram** [comment]
Looks like the 
` User LINUX on system EMBOX1.DEMOS.IBM.COM has been disabled`
error is back on the Actions tests.

**Pull request merged**